File: specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/examples-java/Datastores_Delete.java
import com.azure.core.util.Context;

/** Samples for Datastores Delete. */
public final class Main {
    /*
     * x-ms-original-file: specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/examples/Datastores_Delete.json
     */
    /**
     * Sample code: Datastores_Delete.
     *
     * @param manager Entry point to AvsManager.
     */
    public static void datastoresDelete(com.azure.resourcemanager.avs.AvsManager manager) {
        manager.datastores().delete("group1", "cloud1", "cluster1", "datastore1", Context.NONE);
    }
}

File: specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/examples-js/Datastores_Delete.js
const { AzureVMwareSolutionAPI } = require("@azure/arm-avs");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Delete a datastore in a private cloud cluster
 *
 * @summary Delete a datastore in a private cloud cluster
 * x-ms-original-file: specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/examples/Datastores_Delete.json
 */
async function datastoresDelete() {
  const subscriptionId = "00000000-0000-0000-0000-000000000000";
  const resourceGroupName = "group1";
  const privateCloudName = "cloud1";
  const clusterName = "cluster1";
  const datastoreName = "datastore1";
  const credential = new DefaultAzureCredential();
  const client = new AzureVMwareSolutionAPI(credential, subscriptionId);
  const result = await client.datastores.beginDeleteAndWait(
    resourceGroupName,
    privateCloudName,
    clusterName,
    datastoreName
  );
  console.log(result);
}

datastoresDelete().catch(console.error);
